src/pages/Trade/Trade.tsx

```diff
@@ -18,30 +18,40 @@ import CreateOrderCard from './components/CreateOrderCard';
 import OrderBookCard from './components/OrderBookCard';
 import BlurBackground from './components/BlurBackground';
 
-import { getSynthPair, setSynthPair, SynthPair } from 'ducks/synths';
+import { getSynthPair, setSynthPair, getAvailableSynthsMap } from 'ducks/synths';
 
-type StateProps = {
-	synthPair: SynthPair;
-};
+const mapStateToProps = (state: RootState) => ({
+	synthPair: getSynthPair(state),
+	synthsMap: getAvailableSynthsMap(state),
+});
 
-type DispatchProps = {
-	setSynthPair: typeof setSynthPair;
+const mapDispatchToProps = {
+	setSynthPair,
 };
 
-type Props = RouteComponentProps<{
-	baseCurrencyKey: CurrencyKey;
-	quoteCurrencyKey: CurrencyKey;
-}>;
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
 
-type TradeProps = StateProps & DispatchProps & Props;
+type TradeProps = PropsFromRedux &
+	RouteComponentProps<{
+		baseCurrencyKey: CurrencyKey;
+		quoteCurrencyKey: CurrencyKey;
+	}>;
 
-const Trade: FC<TradeProps> = memo(({ match, setSynthPair, synthPair }) => {
+const Trade: FC<TradeProps> = memo(({ match, setSynthPair, synthPair, synthsMap }) => {
 	const [isReady, setIsReady] = useState<boolean>(false);
 
 	useEffect(() => {
 		const { params } = match;
 
-		if (params && params.baseCurrencyKey && params.quoteCurrencyKey) {
+		if (
+			params &&
+			params.baseCurrencyKey &&
+			params.quoteCurrencyKey &&
+			synthsMap[params.baseCurrencyKey] &&
+			synthsMap[params.quoteCurrencyKey]
```

Comment:
ensure base/quote exists in the synthsMap - in not, redirect to our default synths